lampetia-security-model/src/main/scala/lampetia/security/model/models.scala

```diff
@@ -279,7 +298,7 @@ trait SecurityModel {
       }
       object parentResource extends Composite[Option[Resource]] {
         val resourceId =
-          property[Option[ResourceId]]("resourceId")
```

Comment:
Well, insert accepts a sequence of couples, so you can construct the 
default couples, and then append a sequence of resourceId and 
resourceType or an empty sequence. Something like:

val couples =

    Seq(aclm.id := acl.id.bind, ....)
    ++
    acl.data.parentResource match {
       case None => Seq.empty[Couple]
       case Some(pr) => Seq(aclm.data.parentResource.resourceId :=
    pr.resourceId.bind, ...)
    }

This will avoid the (.get) in the code as well

On 8/7/15 6:40 PM, rosama86 wrote:
>
> In 
> lampetia-security-model/src/main/scala/lampetia/security/model/models.scala 
> <https://github.com/hkarim/lampetia/pull/2#discussion_r36536629>:
>
> > @@ -279,7 +298,7 @@ trait SecurityModel {
> >        }
> >        object parentResource extends Composite[Option[Resource]] {
> >          val resourceId =
> > -          property[Option[ResourceId]]("resourceId")
>
> I think the following code can be improved, but I don't know how yet:
> protected def insertAcl(acl: Acl): IO[Int] = {
> val aclm = AclModel
> acl.data.parentResource match {
> case None =>
> aclm.insert(aclm.id := acl.id.bind,
> aclm.data.permission := acl.data.permission.bind.cast(Types.bit(32)),
> aclm.data.subject.subjectId := acl.data.subject.subjectId.bind,
> aclm.data.subject.subjectType := acl.data.subject.subjectType.bind,
> aclm.data.resource.resourceId := acl.data.resource.resourceId.bind,
> aclm.data.resource.resourceType := acl.data.resource.resourceType.bind)
> case Some(pr) =>
> aclm.insert(aclm.id := acl.id.bind,
> aclm.data.permission := acl.data.permission.bind.cast(Types.bit(32)),
> aclm.data.subject.subjectId := acl.data.subject.subjectId.bind,
> aclm.data.subject.subjectType := acl.data.subject.subjectType.bind,
> aclm.data.resource.resourceId := acl.data.resource.resourceId.bind,
> aclm.data.resource.resourceType := acl.data.resource.resourceType.bind,
> aclm.data.parentResource.resourceId := 
> acl.data.parentResource.get.resourceId.bind,
> aclm.data.parentResource.resourceType := 
> acl.data.parentResource.get.resourceType.bind
> )
> }
> }
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/hkarim/lampetia/pull/2/files#r36536629>.
>